client/views/view_units.cpp

```diff
@@ -17,26 +18,78 @@
 #include "tileset/sprite.h"
 #include "top_bar.h"
 
+// common
+#include "canvas.h"
+#include "goto.h"
+#include "movement.h"
+#include "text.h"
+
 /**
  * Constructor for units view
  */
 units_view::units_view() : QWidget()
 {
   ui.setupUi(this);
 
+  const QPixmap *spr;
+  QImage img;
+  QRect crop;
+  QImage cropped_img;
+  QPixmap pix;
   QStringList slist;
-  slist << _("Unit Type") << _("Upgradable ★") << _("In Progress ⚒")
-        << _("Active ⚔") << _("Shield Upkeep") << _("Food Upkeep")
+
+  // Configure the units table
+  slist << _("Unit Type") << _("★ Upgradable") << _("⚒ In Progress")
+        << _("⚔ Active") << _("Shield Upkeep") << _("Food Upkeep")
         << _("Gold Upkeep") << QLatin1String("");
+  ui.units_label->setText(QString(_("Units:")));
   ui.units_widget->setColumnCount(slist.count());
   ui.units_widget->setHorizontalHeaderLabels(slist);
   ui.units_widget->setSortingEnabled(false);
   ui.upg_but->setText(_("Upgrade"));
   ui.upg_but->setDisabled(true);
   ui.find_but->setText(_("Find Nearest"));
-  ui.find_but->setDisabled(false);
+  ui.find_but->setDisabled(true);
   ui.disband_but->setText(_("Disband All"));
-  ui.disband_but->setDisabled(false);
+  ui.disband_but->setDisabled(true);
+
+  // Configure the unitwaittime table
+  slist.clear();
+  slist << _("Type") << _("Location") << _("Mp") << _("Time left")
+        << QLatin1String("");
+  ui.uwt_widget->setColumnCount(slist.count());
+  ui.uwt_widget->setHorizontalHeaderLabels(slist);
+  ui.uwt_widget->setSortingEnabled(false);
+  ui.uwt_label->setText("Units Waiting:");
+
+  // Add shield icon for shield upkeep column
+  spr = tiles_lookup_sprite_tag_alt(tileset, LOG_VERBOSE, "upkeep.shield",
+                                    "citybar.shields", "", "", false);
+  img = spr->toImage();
+  crop = zealous_crop_rect(img);
+  cropped_img = img.copy(crop);
+  pix = QPixmap::fromImage(cropped_img);
```

Comment:
Ooof we need a shield icon that's not the upkeep one

Reply:
I see that as an artwork request for an update post initial merge.